src/grid/cubic.py

```diff
@@ -0,0 +1,582 @@
+            Fourier1 :
+                Assumes function can be expanded in a Fourier series, and then use Gaussian
+                quadrature. Assumes the function is zero at the boundary of the cube.
+
+                .. math::
+                    w_{ijk} = \frac{8}{(M_x + 1) \cdot (M_y + 1) \cdot (M_z + 1)} \bigg[
+                    \bigg(\sum_{p=1}^{M_x} \frac{\sin(ip \pi/(M_x + 1)) (1 - \cos(p\pi)}{p\pi}
+                         \bigg)
+                         \bigg(\sum_{p=1}^{M_y} \frac{\sin(jp \pi/(M_y + 1)) (1 - \cos(p\pi)}{p\pi}
+                         \bigg)
+                         \bigg(\sum_{p=1}^{M_z} \frac{\sin(kp \pi/(M_z + 1)) (1 - \cos(p\pi)}{p\pi}
+                         \bigg)
+                    \bigg]
+            Fourier2 :
+                Alternative weights based on Fourier series. Assumes the function is zero at the
+                boundary of the cube.
+
+                .. math::
+                    w_{ijk} = V^\prime \cdot w_i w_j w_k,
+                    w_i &= \bigg(\frac{2\sin((j - 0.5)\pi) \sin^2(M_x\pi/2)}{M_x^2 \pi} +
+                     \frac{4}{M_x \pi} \sum_{p=1}^{M_x - 1}
+                     \frac{\sin((2j-1)p\pi /n_x) sin^2(p \pi)\}{\pi}bigg)
+
+            Alternative :
+                This does not assume function is zero at the boundary.
+
+            .. math::
+                w_{ijk} = V \cdot \frac{M_x - 1}{M_x} \frac{M_y - 1}{M_y} \frac{M_z - 1}{M_z}
+
+        """
+        if not isinstance(origin, np.ndarray):
+            raise TypeError("origin should be a numpy array.")
+        if not isinstance(axes, np.ndarray):
+            raise TypeError("axes should be a numpy array.")
+        if not isinstance(shape, np.ndarray):
+            raise TypeError("shape should be a numpy array.")
+        if origin.size != 3 or shape.size != 3:
+            raise ValueError("Origin and shape should have size 3.")
+        if axes.shape != (3, 3):
+            raise ValueError(
+                "Axes {0} should be a three by three array.".format(axes.shape)
+            )
+        if np.any(shape <= 0):
+            raise ValueError("In each coordinate, shape should be positive.")
+
+        self._axes = axes
+        self._origin = origin
+        # Make an array to store coordinates of grid points
+        self._points = np.zeros((shape[0] * shape[1] * shape[2], 3))
+        coords = np.array(
+            np.meshgrid(np.arange(shape[0]), np.arange(shape[1]), np.arange(shape[2]))
+        )
+        coords = np.swapaxes(coords, 1, 2)
+        coords = coords.reshape(3, -1)
+        points = coords.T.dot(self._axes) + origin
+
+        # assign the weights
+        weights = self._choose_weight_scheme(weight_type, shape)
+        super().__init__(points, weights, shape)
+
+    @classmethod
+    def from_molecule(cls, atnums, atcoords, spacing=0.2, extension=5.0, rotate=True):
+        r"""Construct a uniform grid given the molecular coordinates."""
+        raise NotImplementedError
```

Comment:
@Ali-Tehrani can you please complete this method? See https://github.com/theochem/chemtools/blob/master/chemtools/utils/cube.py#L98